feat: add EFP (Ethereum Follow Protocol) integration for wallet

caveman-eth opened the pull request:
Adds support for fetching and managing Ethereum Follow Protocol (EFP) followings in the wallet service

Relating to: https://github.com/status-im/status-desktop/issues/18686
and status-desktop PR: https://github.com/status-im/status-desktop/pull/19195

**Screenshot:**
<img width="3512" height="1994" alt="efp" src="https://github.com/user-attachments/assets/6730863c-468c-4111-9d56-d548a7dd55a1" />


**Complete video demo:** 
https://streamable.com/3y4s1f

**Changes:**

**New EFP Client (`services/wallet/thirdparty/efp/`):**
- HTTP client for api.ethfollow.xyz
- Pagination support (limit/offset parameters)
- Server-side search via `/searchFollowing` endpoint
- Stats endpoint for total following count
- ENS name and avatar resolution included in responses

**Following Manager (`services/wallet/following/`):**
- Provider coordination layer
- Support for multiple data providers (extensible)
- Error handling and logging

**RPC API (`services/wallet/api.go`):**
- `wallet_GetFollowingAddresses(userAddress, search, limit, offset)` - fetch following addresses with optional search and pagination
- `wallet_GetFollowingStats(userAddress)` - get total following count

**Tests:**
- `client_test.go` - 6 unit tests for HTTP client (pagination, search, stats, error handling)
- `manager_test.go` - 5 unit tests for manager (success cases, provider errors, edge cases)
- Mock generation via gomock for testing

#### API Endpoints Used
- `GET /users/{address}/following?include=ens&limit={limit}&offset={offset}&sort=latest`
- `GET /users/{address}/searchFollowing?include=ens&sort=latest&term={search}`
- `GET /users/{address}/stats`

---

caveman-eth committed:
feat: Add EFP following address support

Introduces Ethereum Follow Protocol (EFP) integration for fetching following addresses and stats. Adds a following manager, EFP client, and related API endpoints, along with tests for both manager and client functionality.

caveman-eth commented:
@Khushboo-dev-cpp @dlipicar @jrainville :)

jrainville reviewed (approved):
Awesome work! Looks very good. 

You'll have to rebase and fix some lint issues I think

jrainville commented:
ah, also commits need to follow conventional commit format, eg `feat: add EFP following address support`

igor-sirotin commented:
@caveman-eth I have sent you an Github invite for collaborator. Please accept and rebase to trigger the CI 🙂

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (50)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/1/) | 2025-11-07 17:18:15 | ~1 min | `macos/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/1/consoleText) |
| :heavy_check_mark: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/1/) | 2025-11-07 17:25:11 | ~7 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-171713-01b599-pr7052.zip) |
| :x: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/1/) | 2025-11-07 17:49:33 | ~32 min | `linux/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/1/consoleText) |
| :x: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/1/) | 2025-11-07 17:49:40 | ~32 min | `linux/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/1/consoleText) |
| :heavy_multiplication_x: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/1/) | 2025-11-07 17:50:46 | ~33 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/1//consoleText) |
| :heavy_multiplication_x: | 01b5993f | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/1/) | 2025-11-07 17:51:05 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/1//consoleText) |
| | | | | | | |
| :x: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/2/) | 2025-11-07 17:19:56 | ~58 sec | `macos/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/2/consoleText) |
| :heavy_check_mark: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/2/) | 2025-11-07 17:35:27 | ~10 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-172518-c00dcd-pr7052.zip) |
| :x: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/2/) | 2025-11-07 18:01:05 | ~11 min | `linux/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/2/consoleText) |
| :x: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/2/) | 2025-11-07 18:01:34 | ~11 min | `linux/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/2/consoleText) |
| :heavy_multiplication_x: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/2/) | 2025-11-07 18:02:02 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/2//consoleText) |
| :heavy_multiplication_x: | c00dcd6d | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/2/) | 2025-11-07 18:02:46 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/2//consoleText) |
| | | | | | | |
| :x: | 84ee4b35 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/3/) | 2025-11-07 18:14:24 | ~1 min | `macos/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/3/consoleText) |
| :x: | 84ee4b35 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/3/) | 2025-11-07 18:14:26 | ~1 min | `linux/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/3/consoleText) |
| :x: | 84ee4b35 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/3/) | 2025-11-07 18:14:27 | ~1 min | `linux/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/3/consoleText) |
| :heavy_multiplication_x: | 84ee4b35 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/3/) | 2025-11-07 18:14:36 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/3//consoleText) |
| :heavy_multiplication_x: | 84ee4b35 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/3/) | 2025-11-07 18:14:41 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/3//consoleText) |
| | | | | | | |
| :x: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/4/) | 2025-11-07 18:15:34 | ~1 min | `macos/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/4/consoleText) |
| :x: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/4/) | 2025-11-07 18:15:41 | ~1 min | `linux/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/4/consoleText) |
| :heavy_multiplication_x: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/4/) | 2025-11-07 18:15:43 | ~58 sec | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/4//consoleText) |
| :x: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/4/) | 2025-11-07 18:15:46 | ~1 min | `linux/status-go` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/4/consoleText) |
| :heavy_multiplication_x: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/4/) | 2025-11-07 18:15:51 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/4//consoleText) |
| :heavy_check_mark: | f9570223 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/4/) | 2025-11-07 18:20:43 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-181438-f95702-pr7052.zip) |
| | | | | | | |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/5/) | 2025-11-07 18:42:03 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-183818-286a3a-pr7052.zip) |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/5/) | 2025-11-07 18:42:09 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-183826-286a3a-pr7052.zip) |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/5/) | 2025-11-07 18:46:10 | ~7 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251107-183843-286a3a-pr7052.zip) |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/5/) | 2025-11-07 18:49:12 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251107-183821-286a3a-pr7052.zip) |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/5/) | 2025-11-07 18:51:21 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/5//consoleText) |
| :heavy_check_mark: | 286a3abe | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/5/) | 2025-11-07 19:08:17 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/6/) | 2025-11-12 23:51:44 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-234807-024232-pr7052.zip) |
| :heavy_check_mark: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/6/) | 2025-11-12 23:52:14 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-234816-024232-pr7052.zip) |
| :heavy_multiplication_x: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/6/) | 2025-11-12 23:54:13 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/6//consoleText) |
| :heavy_check_mark: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/6/) | 2025-11-12 23:55:33 | ~7 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-234832-024232-pr7052.zip) |
| :heavy_check_mark: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/6/) | 2025-11-12 23:59:07 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251112-234811-024232-pr7052.zip) |
| :heavy_check_mark: | 0242326d | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/6/) | 2025-11-13 00:00:44 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/7/) | 2025-11-12 23:59:49 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-235615-f71481-pr7052.zip) |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/7/) | 2025-11-13 00:00:26 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-235625-f71481-pr7052.zip) |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/7/) | 2025-11-13 00:03:38 | ~7 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251112-235637-f71481-pr7052.zip) |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/7/) | 2025-11-13 00:10:14 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251112-235915-f71481-pr7052.zip) |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/7/) | 2025-11-13 00:14:01 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/7//consoleText) |
| :heavy_check_mark: | f7148137 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/7/) | 2025-11-13 00:24:45 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/8/) | 2025-11-19 09:57:55 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251119-095404-3b3c0f-pr7052.zip) |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/8/) | 2025-11-19 09:58:18 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251119-095413-3b3c0f-pr7052.zip) |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/8/) | 2025-11-19 10:01:38 | ~7 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251119-095428-3b3c0f-pr7052.zip) |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/8/) | 2025-11-19 10:05:20 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251119-095407-3b3c0f-pr7052.zip) |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/8/) | 2025-11-19 10:06:58 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/8//consoleText) |
| :heavy_check_mark: | 3b3c0f81 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/8/) | 2025-11-19 10:21:12 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3d463fe7 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/9/) | 2025-11-21 17:36:56 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-173230-3d463f-pr7052.zip) |
| :heavy_check_mark: | 3d463fe7 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/9/) | 2025-11-21 17:39:09 | ~6 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-173244-3d463f-pr7052.zip) |
| :heavy_check_mark: | 3d463fe7 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/9/) | 2025-11-21 17:52:18 | ~19 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-173220-3d463f-pr7052.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/10/) | 2025-11-21 17:54:00 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-174919-ba85ed-pr7052.zip) |
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/10/) | 2025-11-21 17:56:28 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-175229-ba85ed-pr7052.zip) |
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/10/) | 2025-11-21 17:58:11 | ~8 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251121-174935-ba85ed-pr7052.zip) |
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/10/) | 2025-11-21 18:01:54 | ~12 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251121-174921-ba85ed-pr7052.zip) |
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/10/) | 2025-11-21 18:04:26 | ~14 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/10//consoleText) |
| :heavy_check_mark: | ba85eddb | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/10/) | 2025-11-21 18:16:59 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/10//consoleText) |
| :x: | ba85eddb | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/nwaku/job/PR-7052/1/) | 2025-12-01 11:18:39 | ~8 min | `macos/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/nwaku/job/PR-7052/1/consoleText) |
| :x: | ba85eddb | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-7052/1/) | 2025-12-01 11:38:16 | ~12 min | `windows/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-7052/1/consoleText) |
| :heavy_check_mark: | ba85eddb | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-7052/1/) | 2025-12-01 13:10:44 | ~57 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-7052/1//consoleText) |
| :heavy_multiplication_x: | ba85eddb | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc-nwaku/job/PR-7052/1/) | 2025-12-01 13:38:06 | ~1 hr 8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc-nwaku/job/PR-7052/1//consoleText) |
| | | | | | | |
| :x: | ad932c1b | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/nwaku/job/PR-7052/2/) | 2025-12-02 21:13:34 | ~2 min | `macos/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/nwaku/job/PR-7052/2/consoleText) |
| :heavy_check_mark: | ad932c1b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-7052/11/) | 2025-12-02 21:14:58 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251202-211109-ad932c-pr7052.zip) |
| :heavy_check_mark: | ad932c1b | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-7052/11/) | 2025-12-02 21:16:54 | ~5 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251202-211123-ad932c-pr7052.zip) |
| :heavy_check_mark: | ad932c1b | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-7052/11/) | 2025-12-02 21:19:56 | ~8 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251202-211151-ad932c-pr7052.zip) |
| :x: | ad932c1b | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-7052/2/) | 2025-12-02 21:21:05 | ~9 min | `windows/nwaku` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-7052/2/consoleText) |
| :heavy_check_mark: | ad932c1b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-7052/11/) | 2025-12-02 21:22:13 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251202-211112-ad932c-pr7052.zip) |
| :heavy_check_mark: | ad932c1b | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-7052/2/) | 2025-12-02 21:38:29 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-7052/2//consoleText) |
| :heavy_check_mark: | ad932c1b | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/12/) | 2025-12-03 00:34:50 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-7052/12//consoleText) |
| :heavy_check_mark: | ad932c1b | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/12/) | 2025-12-03 00:48:04 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-7052/12//consoleText) |

caveman-eth committed:
fix: mockgen command in efp types.go

Changed the go:generate directive to explicitly use 'go tool mockgen' instead of just 'mockgen' for generating mocks. This may improve compatibility or clarity in environments where the tool is not globally installed.

caveman-eth commented:
> @caveman-eth I have sent you an Github invite for collaborator. Please accept and rebase to trigger the CI 🙂

Great, looks like everything is passing now

> Awesome work! Looks very good.
> 
> You'll have to rebase and fix some lint issues I think

@igor-sirotin I think everything is right now :)

dlipicar commented on services/wallet/following/manager.go:
```diff
@@ -0,0 +1,96 @@
+
+// Manager handles following address operations using EFP providers
+type Manager struct {
+	providers []efp.FollowingDataProvider
```
we use a slice of providers for cases where multiple providers can be used to fetch the same type of data. I'm not super well versed in EFP, but it doesn't look like that's something we expect to happen, right?
Moreover, we're just using `m.providers[0]` in all methods, so making this a slice is misleading, better just make it `provider efp.FollowingDataProvider`

dlipicar commented on services/wallet/following/manager.go:
```diff
@@ -0,0 +1,96 @@
+	)
+
+	if len(m.providers) == 0 {
+		return []efp.FollowingAddress{}, nil
```
feels like we should return an error if the provider is not set and we make a request, otherwise we can't distinguish this from a "true" empty FollowingAddresses result.

dlipicar commented on services/wallet/service.go:
```diff
@@ -246,6 +248,13 @@ func NewService(
 
+	// EFP (Ethereum Follow Protocol) providers
+	efpClient := efp.NewClient()
+	followingProviders := []efp.FollowingDataProvider{
```
Check how we conditionally populate the other provider lists https://github.com/status-im/status-go/blob/develop/services/wallet/service.go#L119
Please do the same thing here, we don't want accesses to the provider if `thirdpartyServicesEnabled` is false

dlipicar commented on services/wallet/thirdparty/efp/client.go:
```diff
@@ -0,0 +1,175 @@
+func NewClient() *Client {
+	httpClient := thirdparty.NewHTTPClient(
+		thirdparty.WithDetailedTimeouts(
+			5*time.Second,  // dialTimeout
```
these seem to be opinionated default values without any way of overriding them. Perhaps pass the `httpClient` as a constructor parameter instead?

dlipicar commented on services/wallet/thirdparty/efp/client.go:
```diff
@@ -0,0 +1,175 @@
+const baseURL = "https://api.ethfollow.xyz/api/v1"
+
+const (
+	requestDelay = 100 * time.Millisecond
```
Is this used anywhere?

dlipicar reviewed (commented):
Thanks for your contribution! A couple minor remarks added

igor-sirotin commented on services/wallet/following/manager.go:
```diff
@@ -0,0 +1,96 @@
+// FetchFollowingAddresses fetches the list of addresses that the given user is following
+// Uses the first available provider (can be enhanced later with fallback logic)
+func (m *Manager) FetchFollowingAddresses(ctx context.Context, userAddress common.Address, search string, limit, offset int) ([]efp.FollowingAddress, error) {
+	logutils.ZapLogger().Debug("following.Manager.FetchFollowingAddresses",
```
Would be nice to pass a `logger` to `NewManager`. This way we'll be able to utilize logging namespaces.

Then use it like this everywhere:
```suggestion
	m.logger.Debug("following.Manager.FetchFollowingAddresses",
```

igor-sirotin commented on services/wallet/following/manager_test.go:
```diff
@@ -0,0 +1,116 @@
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	ctx := context.TODO()
```
```suggestion
	ctx := t.Context()
```

igor-sirotin replied:
(and in other tests below)

igor-sirotin commented on services/wallet/thirdparty/efp/client_test.go:
```diff
@@ -0,0 +1,154 @@
+	}
+
+	userAddress := common.HexToAddress("0x742d35cc6cf4c7c7")
+	addresses, err := efpClient.FetchFollowingAddresses(context.Background(), userAddress, "", 10, 0)
```
```suggestion
	addresses, err := efpClient.FetchFollowingAddresses(t.Context(), userAddress, "", 10, 0)
```

igor-sirotin commented on services/wallet/thirdparty/efp/client_test.go:
```diff
@@ -0,0 +1,154 @@
+		},
+	}
+
+	response, _ := json.Marshal(expected)
```
```suggestion
	response, err := json.Marshal(expected)
	require.NoError(err)
```

igor-sirotin commented on services/wallet/thirdparty/efp/client_test.go:
```diff
@@ -0,0 +1,154 @@
+	"github.com/status-im/status-go/services/wallet/thirdparty"
+)
+
+func setupTest(t *testing.T, response []byte) (*httptest.Server, func()) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		_, err := w.Write(response)
+		if err != nil {
+			return
+		}
+	}))
+
+	return srv, func() {
+		srv.Close()
+	}
+}
```
Maybe implement a test suite, not to repeat this call in every test?
```go
type EFPClientTestSuite struct {
    suite.Suite
}

func (s *EFPClientTestSuite) SetupTest() {
     // setup
}

func (s *EFPClientTestSuite) TeardownTest() {
    // cleanup
}
```

igor-sirotin reviewed (approved):
I agree with comments from @dlipicar, otherwise looks good! Thanks for your contribution!
I've left some minor suggestions, no blockers from my side.

caveman-eth committed:
fix(efp): use single EFP provider, initialization test changes, etc

Refactored the following.Manager to accept a single EFP provider and logger instead of a slice of providers. Updated initialization in service.go and all related tests. Adjusted EFP client to require an HTTP client on construction, improving dependency injection and testability.

caveman-eth commented:
@dlipicar and @igor-sirotin thanks for reviewing.
@dlipicar I have now implemented all of your suggestions, thanks for pointing out.
@igor-sirotin I have added all of your suggestions, aside from adding test suite. I could still do it, but didn't put it as high priority.

igor-sirotin reviewed (approved):
Thanks for the fixes 👍

caveman-eth commented:
I have updated the branch with latest upstream commits, and resolved a status-go conflict.

igor-sirotin commented:
@caveman-eth let's make sure https://github.com/status-im/status-desktop/pull/19195 is green and approved, I'll merge this one then

caveman-eth committed:
fix(efp): change api endpoint domain
